### PR TITLE
Block macos fix

### PR DESCRIPTION
**Pull request by mattiashagstrand**
This PR fixes a bug in Block that caused strange rendering on macOS because of the flipped y axis.

### CODE REVIEW

File: Sources/Core/API/Block.swift [hunk 1]
```diff
@@ -119,16 +119,35 @@ public final class Block: SceneObject, InstanceHashable, ActionPerformer, ZIndex
         layer.addSublayer(segmentLayers.top)
         layer.addSublayer(segmentLayers.bottom)
 
+        #if os(macOS)
+        segmentLayers.bottom.frame.origin.y = 0
```
Comment (JohnSundell) [inline Sources/Core/API/Block.swift]: Ah, of course! 🤦‍♂️ Great that you found this @mattiashagstrand 👍

File: Sources/Core/API/Block.swift [hunk 2]
```diff
@@ -119,16 +119,35 @@ public final class Block: SceneObject, InstanceHashable, ActionPerformer, ZIndex
...
+        #else
         segmentLayers.bottom.frame.origin.y = size.height - segmentLayers.bottom.frame.height
+        #endif
 
         let centerReplicatorLayer = ReplicatorLayer()
+        #if os(macOS)
```
Comment (JohnSundell) [inline Sources/Core/API/Block.swift]: All this conditional compilation makes this code really hard to follow IMO (it was already a bit hard to follow, and that's my fault 😅), how about breaking it up into some helper functions and put the conditional compilation inside those functions (will make it a bit more contained)?

Comment (mattiashagstrand) [reply Sources/Core/API/Block.swift]: Absolutely agree with you @JohnSundell, I fell for the old "it's already a bit complicated so what does it matter if I make it a little bit more complicated". My bad! I will take another look and see if I can break up the code a little. It is a little risky since we don't have that many unit tests on Block and it isn't that easy to add either. But at least you get clear visual feedback if it is wrong... 😄

Comment (JohnSundell) [reply Sources/Core/API/Block.swift]: Haha yeah, it's a bit tricky to test these purely visual things 🤔  We could always verify the layer hierarchy but I'm not sure if that makes a ton of sense.

**Comment by mattiashagstrand** [comment]
@JohnSundell, do you think this refactoring is enough?

**Comment by JohnSundell** [comment]
Think I'll make a new release once this is in, we have lots of good stuff and fixes that haven't been shipped yet 😄

**Comment by JohnSundell** [comment]
I think we need to find a CI solution other than Travis 😅

**Comment by mattiashagstrand** [comment]
Yeah, Travis build is still in progress! 😳

**Comment by mattiashagstrand** [comment]
Something must have gone wrong with the Travis build.
@JohnSundell, is there any way to restart the build?

**Comment by JohnSundell** [comment]
@mattiashagstrand I'm just gonna cut Travis off for now, we still have BuddyBuild until we can find a better solution, cuz this ain't working 😅
